lib/app/features/chat/providers/private_direct_chat_provider.c.dart

```diff
@@ -0,0 +1,380 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:cryptography/cryptography.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:ion/app/exceptions/exceptions.dart';
+import 'package:ion/app/features/chat/model/entities/private_direct_message_data.c.dart';
+import 'package:ion/app/features/chat/model/entities/private_message_reaction_data.c.dart';
+import 'package:ion/app/features/core/model/media_type.dart';
+import 'package:ion/app/features/core/providers/env_provider.c.dart';
+import 'package:ion/app/features/nostr/model/entity_expiration.c.dart';
+import 'package:ion/app/features/nostr/model/file_alt.dart';
+import 'package:ion/app/features/nostr/model/nostr_entity.dart';
+import 'package:ion/app/features/nostr/providers/nostr_event_signer_provider.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_notifier.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_upload_notifier.c.dart';
+import 'package:ion/app/services/compressor/compress_service.c.dart';
+import 'package:ion/app/services/logger/logger.dart';
+import 'package:ion/app/services/media_service/media_service.c.dart';
+import 'package:ion/app/services/nostr/ion_connect_gift_wrap_service.c.dart';
+import 'package:ion/app/services/nostr/ion_connect_seal_service.c.dart';
+import 'package:nostr_dart/nostr_dart.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'private_direct_chat_provider.c.g.dart';
+
+@Riverpod(keepAlive: true)
+Future<PrivateDirectChatService> privateDirectChatService(Ref ref) async {
+  final eventSigner = await ref.read(currentUserNostrEventSignerProvider.future);
+
+  if (eventSigner == null) {
+    throw EventSignerNotFoundException();
+  }
+
+  return PrivateDirectChatService(
+    eventSigner: eventSigner,
+    env: ref.read(envProvider.notifier),
+    sealService: ref.read(ionConnectSealServiceProvider),
+    wrapService: ref.read(ionConnectGiftWrapServiceProvider),
+    compressionService: ref.read(compressServiceProvider),
+    nostrNotifier: ref.read(nostrNotifierProvider.notifier),
+    nostrUploadNotifier: ref.read(nostrUploadNotifierProvider.notifier),
+  );
+}
+
+class PrivateDirectChatService {
```

Comment:
Your `PrivateDirectChatService` currently handles multiple responsibilities (compression, encryption, tag generation, orchestrating message sending, etc.), making it quite large and harder to maintain. I would advise splitting it into smaller, dedicated classes, for example:

`PrivateDirectChatOrchestrator` – Coordinates sending messages, delegating compression, encryption, and tag creation to other services.
`MediaCompressionService` – Handles only the logic for compressing media files.
`MediaEncryptionService` – Focuses on encrypting media files.
`TagGenerator` – Responsible for generating conversation and imeta tags.

Reply:
Nice idea, will discuss on daily